### DISCUSSION TITLE
Make Turnkey scripts VLAN compatible (possibly with an env var?)

### DISCUSSION THREAD

**Discussion by rcmurphy** [tteck/Proxmox#1795] [Ideas]
As my setup requires a VLAN to be specified, none of the scripts that have deferred networking configuration until after setup function properly for me. Seemingly this is primarily the Turnkey scripts.

Reading an environment variable such that the new machine is created on a VLAN would be amazing; it would alleviate the requirement of going through advanced setup on non Turnkey VMs too. (e.g. `export LXC_DEFAULT_VLAN=10`)

Thanks!
Rebecca

**Comment by tteck** [comment]
As stated, My intention is to maintain the TurnKey scripts in their simplest form, contained within a single file, and with no options.
You can download the script and locally execute it after modifying line 21 to add the VLAN.
```
-net0 name=eth0,bridge=vmbr0,ip=dhcp,tag=10
```

**Reply by rcmurphy** [reply]
Apologies for misunderstanding your intent; thanks for providing a fix anyway!

**Reply by tteck** [reply]
I've created a `turnkey.sh` script designed to manage the creation of TurnKey LXC containers.

Now users can download the script, customize it as necessary, and then utilize it to create any TurnKey container from a single script, simplifying the process.

Download (all executed within the Proxmox shell):
```
wget -q https://github.com/tteck/Proxmox/raw/main/turnkey/turnkey.sh
```
Edit:
```
nano turnkey.sh
```
Execute:
```
bash turnkey.sh
```
